plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryMetadata.java

```diff
@@ -185,15 +185,23 @@ public List<SchemaTableName> listTables(ConnectorSession session, Optional<Strin
         BigQueryClient client = bigQueryClientFactory.create(session);
 
         log.debug("listTables(session=%s, schemaName=%s)", session, schemaName);
-        String projectId = client.getProjectId();
+        String projectId;
 
+        Set<String> remoteSchemaNames;
         // filter ambiguous schemas
-        Optional<String> remoteSchema = schemaName.flatMap(schema -> client.toRemoteDataset(projectId, schema)
-                .filter(dataset -> !dataset.isAmbiguous())
-                .map(RemoteDatabaseObject::getOnlyRemoteName));
-
-        Set<String> remoteSchemaNames = remoteSchema.map(ImmutableSet::of)
-                .orElseGet(() -> ImmutableSet.copyOf(listRemoteSchemaNames(session)));
+        if (schemaName.isPresent()) {
+            DatasetId localDatasetId = client.toDatasetId(schemaName.get());
+            projectId = localDatasetId.getProject();
+            Optional<String> remoteSchema = client.toRemoteDataset(localDatasetId)
```

Comment:
The above comment "// filter ambiguous schemas" should be moved here.

Reply:
Done!